runtime/src/bank.rs

```diff
@@ -2216,6 +2220,17 @@ impl Bank {
         // although there is no such sysvars currently.
         self.adjust_sysvar_balance_for_rent(&mut new_account);
         self.store_account_and_update_capitalization(pubkey, &new_account);
+
+        if update_cache {
```

Comment:
Would it hurt to update the cache every time any of the sysvars are updated?

Reply:
Slot history and rewards aren't in the cache, so if we try to update the cache with them, the cache yells.  I can change it to have the cache silently fail and eliminate this flag, but I wanted it to be explicit.

Reply:
Mm fair, are you suggesting to do nothing silently then for unsupported sysvars?

Reply:
> are you suggesting to do nothing silently then for unsupported sysvars?

I think `SysvarCache::set_account` can just silently ignore if it does not cache that particular sysvar.

We don't expect a lock contention on `SysvarCache` here, do we?
If there is a possibility of lock contention then, maybe a class method like `SysvarCache::should_cache(...)` to call first before locking the cache.